app/src/main/java/com/github/sdp_begreen/begreen/fragments/ProfileDetailsFragment.kt

```diff
@@ -50,48 +59,228 @@ class ProfileDetailsFragment : Fragment() {
+            if (connectedUserViewModel.currentUser.value?.id == user?.id) {
+                val img = it ?: BitmapFactory.decodeResource(resources, R.drawable.blank_profile_picture)
+                profileImgView.setImageBitmap(BitmapsUtils.rescaleImage(img, 400, 400))
+            } else {
+                lifecycleScope.launch {
+                    val img = user?.let { user ->
+                        user.profilePictureMetadata?.let { pMetadata->
+                            FirebaseDB.getUserProfilePicture(pMetadata, user.id)
+                        }
+                    } ?: BitmapFactory.decodeResource(resources, R.drawable.blank_profile_picture)
+                    profileImgView.setImageBitmap(BitmapsUtils.rescaleImage(img, 400, 400))
+                }
+            }
+        }
+    }
+
+    /**
+     * Helper function to setup the edit button
+     *
+     * Setup if visible or not
+     * setup its listener
+     */
+    private fun setupEditButton(editButton: Button) {
+
+        // listen for currentUserChange to hide or show button accordingly
+        connectedUserViewModel.currentUser.observe(viewLifecycleOwner) {
+            if (it != user) {
+                editButton.visibility = View.GONE
+            }
+        }
+
+        editButton.setOnClickListener {
+            toggleVisibleElement(View.GONE, View.VISIBLE)
+
+            setUpRelativeLayoutAttribute(R.id.fragment_profile_details_save_profile)
+        }
+    }
+
+    /**
+     * Helper function to setup the save button
+     */
+    private fun setupSaveButton(saveButton: Button) {
+        connectedUserViewModel.currentUser.observe(viewLifecycleOwner) {
+            if (it != user) {
+                saveButton.visibility = View.GONE
+            }
+        }
+
+        saveButton.setOnClickListener {
+            toggleVisibleElement(View.VISIBLE, View.GONE)
+            setUpRelativeLayoutAttribute(R.id.fragment_profile_details_edit_profile)
+
+        }
+    }
+
+    /**
+     * Helper function to toggle all element view depending on whether we are in edit mode,
+     * or if we are on "display" mode.
+     *
+     * @param editVisibility The visibility to pass to object that should be visible in edit mode
+     * @param saveVisibility The visibility to pass to object that should be visible in display mode
+     */
+    private fun toggleVisibleElement(editVisibility: Int, saveVisibility: Int) {
+        requireView().findViewById<Button>(R.id.fragment_profile_details_edit_profile)
+            .visibility = editVisibility
+        requireView().findViewById<Button>(R.id.fragment_profile_details_save_profile)
+            .visibility = saveVisibility
+        requireView().findViewById<ImageView>(R.id.fragment_profile_details_profile_image)
+            .visibility = editVisibility
+        requireView().findViewById<ImageButton>(R.id.fragment_profile_details_take_picture)
+            .visibility = saveVisibility
+    }
```

Comment:
Yes great idea, it will indeed be easier to add more component later, by simply add their idea to the list of idea related to either the `edit` or `save` display